parity/upgrade.rs

```diff
@@ -205,6 +201,10 @@ fn upgrade_user_defaults(dirs: &DatabaseDirectories) {
 }
 
 pub fn upgrade_data_paths(base_path: &str, dirs: &DatabaseDirectories, pruning: Algorithm) {
+	if env::home_dir().is_none() {
+		return;
```

Comment:
Does it make make sense to print a warning here?

Reply:
I don't think so. The function only does something if `$HOME/.parity` exists, and doesn't print any warning if it doesn't. If `$HOME` isn't defined we behave exactly as if `$HOME/.parity` doesn't exist.